pkg/gateway/lb/handler/grpc/handler.go

```diff
@@ -1107,6 +1174,46 @@ func (s *server) linearsearch(ctx context.Context, cfg *payload.Search_Config,
 			if num != 0 && len(res.GetResults()) > num {
 				res.Results = res.GetResults()[:num]
 			}
+
+			if errors.Is(ectx.Err(), context.DeadlineExceeded) {
+				if len(res.GetResults()) == 0 {
+					err = status.WrapWithDeadlineExceeded(
+						"error search result length is 0",
+						errors.ErrEmptySearchResult,
+						&errdetails.RequestInfo{
+							RequestId:   cfg.GetRequestId(),
+							ServingData: errdetails.Serialize(cfg),
+						},
+						&errdetails.ResourceInfo{
+							ResourceType: errdetails.ValdGRPCResourceTypePrefix + "/vald.v1.search",
+							ResourceName: fmt.Sprintf("%s: %s(%s) to %v", apiName, s.name, s.ip, s.gateway.Addrs(ctx)),
+						}, info.Get(),
+					)
+					if span != nil {
+						span.SetStatus(trace.StatusCodeDeadlineExceeded(err.Error()))
+					}
+					return nil, err
+				}
+				if min != 0 && len(res.GetResults()) < min {
+					err = status.WrapWithDeadlineExceeded(
+						"error search result length less than minimum number",
```

Comment:
How about showing the real result length?
It is more convenient for the user.

Reply:
same
https://github.com/vdaas/vald/pull/1546#discussion_r802291404